File: gntl_app_backend/src/main/java/com/gentlemanstore/product/dto/ProductDTO.java
package com.gentlemanstore.product.dto;

import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductDTO {
    private Long id;
    private String sku;
    private String name;
    private String description;
    private BigDecimal price;
    private String categoryName;
    private List<ProductSizeDTO> sizes;
    private List<String> imageUrls;
    private List<String> tags;
    private BigDecimal discountPercentage;
    // Staff prikaz DELETED/ALL - UI mora znati da li je proizvod obrisan
    private boolean deleted;
}

File: gntl_app_frontend/app/src/main/java/com/gentlemanstore/feature/product/data/dto/ProductDto.kt
package com.gentlemanstore.feature.product.data.dto

data class ProductSizeResponse(
    val id: Long,
    val size: String,
    val quantity: Int
)

data class ProductResponse(
    val id: Long,
    val sku: String,
    val name: String,
    val description: String,
    val price: Double,
    val categoryName: String,
    val sizes: List<ProductSizeResponse>,
    val imageUrls: List<String>,
    val tags: List<String>,
    val discountPercentage: Double?,
    // Staff DELETED/ALL prikaz - obrisan proizvod se oznacava i dobija Restore
    val deleted: Boolean = false
)

data class PagedProductResponse(
    val content: List<ProductResponse>,
    val totalPages: Int,
    val totalElements: Long,
    val last: Boolean,
    val first: Boolean,
    val number: Int,
    val size: Int,
    val numberOfElements: Int
)

data class CategoryResponse(
    val id: Long,
    val name: String,
    val description: String?
)

// Employee CRUD — odgovara backend CreateProductRequest (create i update)
data class SizeRequest(
    val size: String,
    val quantity: Int
)

data class CreateProductRequest(
    val sku: String,
    val name: String,
    val description: String,
    val price: Double,
    val categoryId: Long,
    val sizes: List<SizeRequest>,
    val imageUrls: List<String>,
    val tags: List<String>
)